### DISCUSSION TITLE
Finding matched index while using array matcher

### DISCUSSION THREAD

**Discussion by byucesoy** [jeremyevans/roda#362] [General]
Arrays match when any of their elements match. Is there a way to find which element matched to the path?

**Comment by jeremyevans** [comment]
If the matching element is a string, it is yielded.  For other cases, it's probably not possible to know.  You can use an iterative approach in that case:

```ruby
  array.each do |element|
    r.on element do
      # ...
    end
  end
```

**Comment by byucesoy** [comment]
In the API I'm designing, it is possible to access resources in two ways: `/id/<resource-id>` or `/<resource-name>`. In both cases, the business logic is quite similar apart from the initial lookup, so I'm doing something like this.

```
r.on [{all: "id", String}, String] do |resource_identifier|
```

So the matching element is string, but it is not possible to differentiate the index based on the value of the yielded value. Iterative approach you suggested makes sense. I'll go with that. Thanks.
